src/app/features/notifications/notifications.component.ts

```diff
@@ -74,16 +66,15 @@ export class NotificationsComponent implements OnInit {
    * Loads all notifications and processes them to update the list and unread count.
    */
   private async loadAndProcessNotifications(): Promise<void> {
-    const allNotifications =
+    const notifications =
       await this.entityMapper.loadType<NotificationActivity>(
         NotificationActivity,
       );
     // The user is hardcoded for testing purposes, need to remove this.
-    this.notifications = this.filterUserNotifications(
-      allNotifications,
-      "User:demo",
+    this.filterUserNotifications(notifications, "User:demo");
+    this.hasUnreadNotificationCount = this.countUnreadNotifications(
+      this.allNotifications,
     );
```

Comment:
This is to get the dynamic notification list corresponding to the user. But the `this.currentUser.value?.getId()` getting undefined that's why I have added the hardcoded value just for testing purposes. I will remove this when I implement the notification list logic.

Reply:
I think in our case we should work with `SessionInfo` and its `id` property to identify and filter for users. Every authenticated user will have this (it's the keycloak user account ID).

While the demo should give you a defined user entity with `CurrentUserSubject` (not sure why you are getting undefined ...), this is not available for every user in every system, depending on the custom setup.